### PR TITLE
Shirley flashcard

### PR DESCRIPTION
**Pull request by shirleymango**
Created a flashcard with text
1 text on the screen

Added horizontal flip to flashcard
https://user-images.githubusercontent.com/44847817/178074082-5137024c-55c8-4724-b1ca-0bab6a0a5454.mp4

Added flip with text staying in place
https://user-images.githubusercontent.com/44847817/178074089-7899b294-3f27-4e31-bbdb-0efda39207e7.mp4

Created back side of the card with flip
https://user-images.githubusercontent.com/44847817/178074095-6a9cd7a3-036a-4bcc-8ff7-385dd95a8659.mp4

Put the two sides together, changing the z position so they alternate being on top
https://user-images.githubusercontent.com/44847817/178074101-8bdac86d-42f4-40be-9a03-3c0b40384b86.mp4

- the front and back sides of the card are each a CALayer
- added text with a CATextLayer
- created flip animations with CATransform3DMakeRotation and CATransform3DRotate
- created a boolean isFlipped so opposite animation can happening depending on boolean value
- created tap gesture and added animation to gesture action

### CODE REVIEW

**Comment by shirleymango** [comment]
https://user-images.githubusercontent.com/44847817/178074579-065b492d-af5f-4b5e-94de-06d71286d309.mp4

File: metau-capstone/StudyViewController.m [hunk 1]
```diff
@@ -10,14 +10,79 @@
...
+    self.back.position = CGPointMake(self.view.center.x, self.view.center.y - 50);
+    
+    // add text label to the flashcard
+    CATextLayer *backLabel = [[CATextLayer alloc] init];
+    [backLabel setFont:@"Helvetica-Bold"];
+    [backLabel setFontSize:20];
+    [backLabel setString:@"Back"];
+    [backLabel setAlignmentMode:kCAAlignmentCenter];
+    [backLabel setForegroundColor:[[UIColor whiteColor] CGColor]];
+    [backLabel setFrame:CGRectMake(0, 0, 300, 180)];
```
Comment (zenangao-fb) [inline metau-capstone/StudyViewController.m:37]: It seems a lot of common properties set both for front and back, considering create a helper function. DRY

File: metau-capstone/StudyViewController.m [hunk 2]
```diff
@@ -10,14 +10,79 @@
...
+    [label setString:@"Front"];
+    [label setAlignmentMode:kCAAlignmentCenter];
+    [label setForegroundColor:[[UIColor blackColor] CGColor]];
+    [label setFrame:CGRectMake(0, 0, 300, 180)];
+    [self.front addSublayer:label];
+    
+    [self.view.layer addSublayer:self.front];
+    
+    // create rotation animation
+    self.rotateAnim = [CABasicAnimation animationWithKeyPath:@"transform.rotation.y"];
```
Comment (zenangao-fb) [inline metau-capstone/StudyViewController.m:61]: I noticed some flickering during the flip animation. Have you tried using UIView.transition function instead?
example in swift: 
https://github.com/Sajjon/SwiftIntro/blob/master/SwiftIntro/Views/Cells/CollectionViewCells/CardCVCell/CardCVCell.swift

**Comment by zenangao-fb** [comment]
Some Edge use case, will the card view be scrollable when the content cannot fit for the designed area?

**Pull request merged**